common/persistence/history_manager.go

```diff
@@ -527,6 +527,17 @@ func (m *executionManagerImpl) AppendRawHistoryNodes(
 	}, err
 }
 
+func (m *executionManagerImpl) NewHistoryBranch(
+	ctx context.Context,
+	request *NewHistoryBranchRequest,
+) (*NewHistoryBranchResponse, error) {
+
+	resp := &NewHistoryBranchResponse{}
+	var err error
+	resp.BranchToken, err = NewHistoryBranchToken(request.TreeId)
```

Comment:
make this private and all initiate history branch should go through execution manager?

Reply:
It's being used by a test which unfortunately is not in the same package (see https://github.com/temporalio/temporal/blob/master/common/persistence/persistence-tests/historyV2PersistenceTest.go#L714)

Reply:
@wxing1292 Is it possible to update the persistencetest package to persistence?

Reply:
I tried and it's ugly. If I move historyV2PersistenceTest.go up one directory, it'll need to import TestBase from the original subdir. Unfortunately, this creates a cyclic import situation because multiple sources in the subdir import from the parent `persistence` package. I'd suppose it's possible to move multiple (if not all) files from the subdir up to the `persistence` package level. But this is probably not worthwhile just to hide this single helper function.

Reply:
nvm, I am wrong.